tracee-ebpf/test/integration/integration_test.go

```diff
@@ -0,0 +1,256 @@
+	// create a temp dir for testing
+	d, err := ioutil.TempDir("", "Test_MagicWrite-dir-*")
+	require.NoError(t, err)
+
+	// cp a file to trigger
+	f, err := os.CreateTemp(d, "Test_MagicWrite-file-*")
+	require.NoError(t, err)
+	defer func() {
+		os.Remove(d)
+	}()
+
+	f.WriteString(`foo.bar.baz`)
+	f.Close()
+
+	cpCmd := exec.Command("cp", f.Name(), filepath.Join(d+filepath.Base(f.Name())+"-new"))
+	fmt.Println("executing: ", cpCmd.String())
+	cpCmd.Stdout = os.Stdout
+	assert.NoError(t, cpCmd.Run())
+
+	// check tracee output
+	assert.Contains(t, gotOutput.String(), `[102 111 111 46 98 97 114 46 98 97 122]`)
+}
+
+// execute a ls command
+func checkExeccommand(t *testing.T, gotOutput *bytes.Buffer) {
+	_, _ = exec.Command("ls").CombinedOutput()
+
+	// check tracee output
+	processNames := strings.Split(strings.TrimSpace(gotOutput.String()), "\n")
+	for _, pname := range processNames {
+		assert.Equal(t, "ls", pname)
+	}
+}
+
+// only capture new pids after tracee
+func checkPidnew(t *testing.T, gotOutput *bytes.Buffer) {
+	traceePid := getPidByName(t, "tracee")
+
+	// run a command
+	_, _ = exec.Command("ls").CombinedOutput()
+
+	// output should only have events with pids greater (newer) than tracee
+	pids := strings.Split(strings.TrimSpace(gotOutput.String()), "\n")
+	for _, p := range pids {
+		pid, _ := strconv.Atoi(p)
+		assert.Greater(t, pid, traceePid)
+	}
+}
+
+// only capture uids of 0 that are run by comm ls
+func checkUidzero(t *testing.T, gotOutput *bytes.Buffer) {
+	_, _ = exec.Command("ls").CombinedOutput()
+
+	// check output length
+	require.NotEmpty(t, gotOutput.String())
+
+	// output should only have events with uids of 0
+	uids := strings.Split(strings.TrimSpace(gotOutput.String()), "\n")
+	for _, u := range uids {
+		uid, _ := strconv.Atoi(u)
+		require.Zero(t, uid)
+	}
+}
+
+// only capture pids of 1
+func checkPidOne(t *testing.T, gotOutput *bytes.Buffer) {
+	_, _ = exec.Command("init", "q").CombinedOutput()
+
+	// check output length
+	require.NotEmpty(t, gotOutput.String())
+
+	// output should only have events with pids of 1
+	pids := strings.Split(strings.TrimSpace(gotOutput.String()), "\n")
+	for _, p := range pids {
+		pid, _ := strconv.Atoi(p)
+		require.Equal(t, 1, pid)
+	}
+}
+
+// check that execve event is called
+func checkExecve(t *testing.T, gotOutput *bytes.Buffer) {
+	_, _ = exec.Command("ls").CombinedOutput()
+
+	// check output length
+	require.NotEmpty(t, gotOutput.String())
+
+	// output should only have events with event name of execve
+	eventNames := strings.Split(strings.TrimSpace(gotOutput.String()), "\n")
+	for _, en := range eventNames {
+		if len(en) > 0 {
+			require.Equal(t, "execve", en)
+		}
+	}
+}
+
+// check for filesystem set when ls is invoked
+func checkSetFs(t *testing.T, gotOutput *bytes.Buffer) {
+	_, _ = exec.Command("ls").CombinedOutput()
+
+	// check output length
+	require.NotEmpty(t, gotOutput.String())
+
+	// output should only have events with event name of execve
```

Comment:
Comment should probably be related to set=fs

Reply:
Thanks fixed.